src/CMakeLists.txt

```diff
@@ -226,69 +240,91 @@ register_migraphx_ops(
 )
 register_op(migraphx HEADER migraphx/op/rnn_variable_seq_lens.hpp OPERATORS op::rnn_var_sl_shift_output op::rnn_var_sl_shift_sequence)
 register_op(migraphx HEADER migraphx/builtin.hpp OPERATORS builtin::literal builtin::param builtin::returns)
-rocm_clang_tidy_check(migraphx)
+
+if(CLANG_TIDY_EXE)
+    rocm_clang_tidy_check(migraphx)
+endif()
+
 rocm_install_targets(
   TARGETS migraphx
   INCLUDE
     ${CMAKE_CURRENT_SOURCE_DIR}/include
     ${CMAKE_CURRENT_BINARY_DIR}/include
 )
 
-
-check_cxx_linker_flag(-lstdc++fs HAS_LIB_STD_FILESYSTEM)
-if(HAS_LIB_STD_FILESYSTEM)
-target_link_libraries(migraphx PRIVATE -lstdc++fs)
+if(NOT WIN32)
+    check_cxx_linker_flag(-lstdc++fs HAS_LIB_STD_FILESYSTEM)
+    if(HAS_LIB_STD_FILESYSTEM)
+        target_link_libraries(migraphx PRIVATE -lstdc++fs)
+    endif()
+    target_link_libraries(migraphx PRIVATE -ldl)
 endif()
 
-target_link_libraries(migraphx PRIVATE -ldl)
-
 target_include_directories(migraphx SYSTEM PUBLIC $<BUILD_INTERFACE:${HALF_INCLUDE_DIR}>)
-
-find_package(Threads)
 target_link_libraries(migraphx PUBLIC Threads::Threads)
 
-find_package(nlohmann_json 3.8.0 REQUIRED)
+if(NOT WIN32)
+    find_package(nlohmann_json 3.8.0 REQUIRED)
+endif()
+
 target_link_libraries(migraphx PRIVATE nlohmann_json::nlohmann_json)
 
-find_package(PkgConfig)
-pkg_check_modules(SQLITE3 REQUIRED IMPORTED_TARGET sqlite3)
-target_link_libraries(migraphx PRIVATE PkgConfig::SQLITE3)
+generate_export_header(migraphx
+    EXPORT_FILE_NAME
+        ${CMAKE_BINARY_DIR}/include/migraphx/export.h)
+
+target_include_directories(migraphx PRIVATE ${CMAKE_BINARY_DIR}/include)
+
+if(WIN32)
+    target_link_libraries(migraphx PRIVATE sqlite3::sqlite3 msgpackc-cxx)
+    if(CMAKE_CXX_COMPILER STREQUAL "Clang")
+        target_compile_options(migraphx PRIVATE -fdelayed-template-parsing -fms-extensions -fms-compatibility)
+    endif()
+else()
+    find_package(PkgConfig)
+    pkg_check_modules(SQLITE3 REQUIRED IMPORTED_TARGET sqlite3)
+    target_link_libraries(migraphx PRIVATE PkgConfig::SQLITE3)
 
-find_package(msgpackc-cxx QUIET)
-if(NOT msgpackc-cxx_FOUND)
     find_package(msgpack REQUIRED)
+    target_link_libraries(migraphx PUBLIC msgpackc-cxx)
 endif()
-target_link_libraries(migraphx PRIVATE msgpackc-cxx)
-# Make this available to the tests
-target_link_libraries(migraphx INTERFACE $<BUILD_INTERFACE:msgpackc-cxx>)
-
-add_library(migraphx_all_targets INTERFACE)
 
-set(PACKAGE_DEPENDS)
+add_custom_target(migraphx_all_targets)
```

Comment:
Why is this changed? A custom target cant be linked.

Reply:
I explained below @umangyadav's comment the reason why we changed that target. However, we misunderstood the meaning of the target. We thought it was aggregating target, but making it such had not disturbed the linking, because the target doubled the linking information.